core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackSyncService.java

```diff
@@ -59,11 +62,14 @@
     @javax.annotation.Resource
     private Map<CloudPlatform, MetadataSetup> metadataSetups;
 
-    public void sync(Long stackId) {
+    public void sync(Long stackId) throws Exception {
         Stack stack = stackService.getById(stackId);
+        AmbariClient ambariClient = getAmbariClient(stack);
         Set<InstanceMetaData> instances = instanceMetaDataRepository.findAllInStack(stackId);
         Map<InstanceSyncState, Integer> instanceStateCounts = initInstanceStateCounts();
         for (InstanceMetaData instance : instances) {
+            Map<String, String> hostNames = ambariClient.getHostNames();
+            updateHostMetadataByHostState(stack, ambariClient, instance, hostNames);
```

Comment:
HostMetadata belongs to the cluster layer - updating the host state should belong to the cluster sync layer as well. I think `AmbariClusterFacade.sync` should call a new service (e.g.: `AmbariHostsSyncService`) that will hold this code part.